source/bringauto/common_utils/ProtobufUtils.cpp

```diff
@@ -73,15 +72,15 @@ ExternalProtocol::ExternalClient ProtobufUtils::createExternalClientStatus(const
 																		   ExternalProtocol::Status_DeviceState deviceState,
 																		   u_int32_t messageCounter,
 																		   const InternalProtocol::DeviceStatus &deviceStatus,
-																		   const buffer &errorMessage) {
+																		   const bringauto::modules::Buffer &errorMessage) {
 	ExternalProtocol::ExternalClient externalMessage;
 	ExternalProtocol::Status *status = externalMessage.mutable_status();
 	status->mutable_devicestatus()->CopyFrom(deviceStatus);
 	status->set_sessionid(sessionId);
 	status->set_devicestate(deviceState);
 	status->set_messagecounter(messageCounter);
-	if(errorMessage.size_in_bytes > 0 && errorMessage.data != nullptr) {
-		status->set_errormessage(errorMessage.data, errorMessage.size_in_bytes);
+	if(errorMessage.getStructBuffer().size_in_bytes > 0 && errorMessage.getStructBuffer().data != nullptr) {
```

Comment:
Maybe save erroBessage strus ::buffer into const variable? for better readability....

Reply:
added a new function that replaces this condition